addons/link_tracker/models/mail_render_mixin.py

```diff
@@ -40,19 +40,20 @@ def _shorten_links(self, html, link_tracker_vals, blacklist=None, base_url=None)
             long_url = match[1]
```

Comment:
Doesn't the same issue as in `_shorten_links_text` occur here? If `html` contains multiple occurrences of the same URL, the short-url-ification will be invoked multiple times, unnecessarily.

Reply:
Absolutely!